tests-pending/rpc_pending_test.go

```diff
@@ -0,0 +1,332 @@
+
+	require.Equal(t, uint64(preTxPendingTxCount)+uint64(1), uint64(postTxPendingTxCount))
+	require.NotEqual(t, uint64(postTxPendingTxCount)-uint64(preTxPendingTxCount), uint64(postTxLatestTxCount)-uint64(preTxLatestTxCount))
+}
+
+func TestEth_Pending_GetBlockByNumber(t *testing.T) {
+	rpcRes := util.Call(t, "eth_getBlockByNumber", []interface{}{"latest", true})
+	var preTxLatestBlock map[string]interface{}
+	err := json.Unmarshal(rpcRes.Result, &preTxLatestBlock)
+	require.NoError(t, err)
+	preTxLatestTxs := len(preTxLatestBlock["transactions"].([]interface{}))
+
+	rpcRes = util.Call(t, "eth_getBlockByNumber", []interface{}{"pending", true})
+	var preTxPendingBlock map[string]interface{}
+	err = json.Unmarshal(rpcRes.Result, &preTxPendingBlock)
+	require.NoError(t, err)
+	preTxPendingTxs := len(preTxPendingBlock["transactions"].([]interface{}))
+
+	param := make([]map[string]string, 1)
+	param[0] = make(map[string]string)
+	param[0]["from"] = "0x" + fmt.Sprintf("%x", from)
+	param[0]["to"] = addrA
+	param[0]["value"] = "0xA"
+	param[0]["gasLimit"] = "0x5208"
+	param[0]["gasPrice"] = "0x1"
+
+	_ = util.Call(t, "eth_sendTransaction", param)
+
+	rpcRes = util.Call(t, "eth_getBlockByNumber", []interface{}{"pending", true})
+	var postTxPendingBlock map[string]interface{}
+	err = json.Unmarshal(rpcRes.Result, &postTxPendingBlock)
+	require.NoError(t, err)
+	postTxPendingTxs := len(postTxPendingBlock["transactions"].([]interface{}))
+	require.Greater(t, postTxPendingTxs, preTxPendingTxs)
+
+	rpcRes = util.Call(t, "eth_getBlockByNumber", []interface{}{"latest", true})
+	var postTxLatestBlock map[string]interface{}
+	err = json.Unmarshal(rpcRes.Result, &postTxLatestBlock)
+	require.NoError(t, err)
+	postTxLatestTxs := len(postTxLatestBlock["transactions"].([]interface{}))
+	require.Equal(t, preTxLatestTxs, postTxLatestTxs)
+
+	require.Greater(t, postTxPendingTxs, preTxPendingTxs)
+}
+
+func TestEth_Pending_GetTransactionByBlockNumberAndIndex(t *testing.T) {
+	param := make([]map[string]string, 1)
+	param[0] = make(map[string]string)
+	param[0]["from"] = "0x" + fmt.Sprintf("%x", from)
+	param[0]["to"] = addrA
+	param[0]["value"] = "0xA"
+	param[0]["gasLimit"] = "0x5208"
+	param[0]["gasPrice"] = "0x1"
+
+	_ = util.Call(t, "eth_sendTransaction", param)
+
+	rpcRes := util.Call(t, "eth_getTransactionByBlockNumberAndIndex", []interface{}{"pending", "0x1"})
+	var pendingBlock map[string]interface{}
+	err := json.Unmarshal(rpcRes.Result, &pendingBlock)
+	require.NoError(t, err)
+
+	require.Equal(t, pendingBlock["blockHash"], nil)
+	require.Equal(t, pendingBlock["blockNumber"], nil)
+	require.Equal(t, pendingBlock["transactionIndex"], nil)
+	require.NotEmpty(t, pendingBlock["hash"])
+
+	rpcRes = util.Call(t, "eth_getTransactionByBlockNumberAndIndex", []interface{}{"latest", "0x1"})
+	var latestBlock map[string]interface{}
+	err = json.Unmarshal(rpcRes.Result, &latestBlock)
+	require.NoError(t, err)
+
+	require.Empty(t, latestBlock)
```

Comment:
I'm confused as to what's happening here, I thought `eth_getTransactionByBlockNumberAndIndex` returned a transaction

Reply:
we are checking for the returned transaction here. the check is a bit elementary and may need a more definitive check to make sure that the transaction exists